src/ocm/contexts/FeatureSupportLevelContextProvider.tsx

```diff
@@ -0,0 +1,62 @@
+import useSWR from 'swr';
+import React, { PropsWithChildren } from 'react';
+import FeatureSupportLevelContext, {
+  EmptyFeatureSupportLevelData,
+} from '../../common/contexts/FeatureSupportLevelContext';
+import { getErrorMessage, handleApiError } from '../api/utils';
+import { FeatureSupportLevels, useAlerts } from '../../common';
+import { FeatureSupportLevelsAPI } from '../services/api';
+import FeatureSupportLevelData from './FeatureSupportLevelData';
+
+export type SupportLevelProviderProps = PropsWithChildren<{
+  clusterFeatureUsage?: string;
+  openshiftVersion?: string;
+  loadingUi: React.ReactNode;
+}>;
+
+export const FeatureSupportLevelProvider = ({
+  children,
+  loadingUi,
+  clusterFeatureUsage,
+  openshiftVersion,
+}: SupportLevelProviderProps) => {
+  const alerts = useAlerts();
+  const fetcher = () => FeatureSupportLevelsAPI.list().then((res) => res.data);
```

Comment:
this is a new instance on every rerender. Is it not going to cause problems in `useSWR` hook ? I think it would be safer to move it ouside the component

Reply:
It definitely isn't called on every rerender, useSWR makes sure of that. it does get called however on every onMount. I disabled it for rerunning every focus. I wanted to prevent that by upgrading useSWR to 1.0.1 to use useSWRImmutable that also takes it from the cache, but I saw it was complicated since OCM and CIM will need to be updated. There is this option however: 
1. to add another context in the router of the application that calls this API and stores just the value of the global vsupport level versions and this context will use it
2. not use useSWR at all... So it won't revalidate on mount